test/clients/consumer/messages-stream.test.ts

```diff
@@ -858,6 +859,37 @@ test('should properly handle close', async t => {
   await stream.close()
 })
 
+test.only('should properly handle deleting topics in between', async t => {
+  const admin = createAdmin(t)
+  const topic1 = await createTopic(t)
+  const consumer = createConsumer(t)
+  await admin.createTopics({ topics: [topic1] })
+
+  await consumer.consume({
+    topics: [topic1],
+    mode: MessagesStreamModes.EARLIEST,
+    autocommit: true,
+    maxWaitTime: 1000
+  })
+
+  await admin.deleteTopics({ topics: [topic1] })
+
+  const topic2 = await createTopic(t)
+  await admin.createTopics({ topics: [topic2] })
+
+  await consumer.consume({
+    topics: [topic2],
+    mode: MessagesStreamModes.EARLIEST,
+    autocommit: true,
+    maxWaitTime: 1000
+  })
```

Comment:
I think you need to destroy the resulting stream. There is no automatic finalization.